### PR TITLE
feat: implement overloaded reset method with selective data clearing options

### PR DESCRIPTION
**Pull request by SKannaniOS**
## Description
This pull request implements an enhanced version of the `reset()` method that provides granular control over which user identity data components should be cleared during a reset operation. The implementation adds optional parameters to the existing reset functionality while maintaining backward compatibility.

The new implementation allows users to selectively reset specific data components (anonymous ID, user ID, traits, session) rather than performing a complete reset of all user data. This addresses use cases where applications need to clear only certain user identity information while preserving others.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Code refactor/optimization

## Implementation Details
- **Core Reset Options**: Added `ResetOptions` class that encapsulates reset configuration through `ResetEntries`
- **Granular Reset Control**: Implemented `ResetEntries` class with boolean flags for `anonymousId`, `userId`, `traits`, and `session` components
- **Enhanced Analytics API**: Modified `Analytics.reset()` method to accept optional `ResetOptions` parameter while maintaining backward compatibility
- **State Management Updates**: Updated `ResetUserIdentityAction` to handle selective reset operations based on provided entries
- **User Identity Enhancement**: Enhanced `UserIdentity` class with `resetUserIdentity(storage:entries:)` method for selective data clearing
- **Objective-C Support**: Added `ObjCResetEntriesBuilder` and `ObjCResetOptionsBuilder` classes for fluent API construction in Objective-C environments
- **Session Management**: Conditional session refresh that only occurs when session reset is enabled in options
- **File Organization**: Restructured state management files by moving user identity related actions to dedicated `UserIdentity` subfolder

## Checklist
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation (if appropriate).
- [x] I have ensured that my code follows the project's code style.
- [x] I have checked for potential performance impacts and optimized if necessary.
- [x] I have checked the code for security issues.
- [ ] I have updated the changelog (if required).

## How to test?
The implementation includes comprehensive test coverage across multiple test suites:

### Swift Analytics Tests
- **Default Reset Behavior**: Verifies that calling `reset()` without parameters maintains existing behavior (resets all data)
- **Custom Reset Options**: Tests selective reset functionality with various combinations of reset entries
- **Storage Consistency**: Validates that in-memory state changes are properly persisted to storage
- **Edge Cases**: Tests reset behavior after client shutdown and with empty initial state

### State Management Tests  
- **Action-Level Testing**: Unit tests for `ResetUserIdentityAction` with different `ResetEntries` configurations
- **Selective Reset Validation**: Tests that verify only specified components are reset while others are preserved

### Objective-C Integration Tests
- **Builder Pattern**: Tests for `ObjCResetEntriesBuilder` and `ObjCResetOptionsBuilder` functionality
- **API Compatibility**: Verification that Objective-C wrapper properly bridges to Swift implementation

**Test Environment**: 
- Run the test suite using `swift test` or through Xcode Test Navigator
- Test cases cover iOS, macOS, tvOS, and watchOS platforms
- All tests pass with both in-memory and disk-based analytics configurations

## Breaking Changes
This is a non-breaking change. The existing `reset()` method continues to work exactly as before, resetting all user identity data. The new functionality is additive and accessed through an optional parameter.

## Maintainers Checklist
- [ ] The code has been reviewed.
- [ ] CI tests have passed.
- [ ] All necessary documentation has been updated.

## Screenshots (if applicable)
N/A - This is an API enhancement without UI changes.

## Additional Context
- **Backward Compatibility**: Maintains full compatibility with existing reset usage patterns
- **Architecture**: Follows existing state management patterns and builder design patterns for Objective-C compatibility
- **Future Extensibility**: The `ResetEntries` and `ResetOptions` design allows for easy addition of new resetable components in future versions

The implementation enables use cases such as:
- Clearing user identification while preserving session state
- Resetting traits without affecting user/anonymous IDs  
- Selective anonymous ID regeneration while maintaining user context
- Session refresh without clearing user identity data

## Summary by CodeRabbit

- New Features
  - Granular reset options to control clearing of anonymous ID, user ID, traits, and session.
  - Objective-C–friendly builders and an ObjC-visible reset(options:) API.
  - Example app adds a “Reset with Options” action.

- Refactor
  - Project reorganized to group User Identity and Reset components.

- Chores
  - SDK version updated to 1.0.0-alpha.pr.136.1.

- Tests
  - Expanded tests for selective resets, storage consistency, post-shutdown behavior, and plugin add/remove.

### CODE REVIEW

**Review by 1abhishekpandey — APPROVED** [review]

**Review by 1abhishekpandey — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

File: Tests/RudderStackAnalyticsTests/Helpers/MockProvider.swift [hunk 1]
```diff
@@ -144,21 +144,46 @@ class MockAnalytics: Analytics {
...
+    let nanoseconds = UInt64(seconds * 1_000_000_000)
+    try? await Task.sleep(nanoseconds: nanoseconds)
     
-    func then(_ description: String = "", closure: () -> Void) {
-        if !description.isEmpty { print("Then \(description)") }
-        closure()
-    }
+    // Execute the block after the delay
+    await block()
 }
```
[the remark this answers is not in the document]
Comment (SKannaniOS) [reply Tests/RudderStackAnalyticsTests/Helpers/MockProvider.swift:187]: Will clean that in next PR.
